src/app/admin/components/change-author/components/change-author-user-dropdown/change-author-user-dropdown.component.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Component,
+  OnInit,
+  IterableDiffers,
+  IterableDiffer,
+  EventEmitter,
+  Output,
+} from '@angular/core';
+import { User } from '@entity';
+import { UserService } from 'app/core/user.service';
+import { AuthService } from 'app/core/auth.service';
+import { Subject } from 'rxjs';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+ 
+@Component({
+  selector: 'clark-change-author-user-dropdown',
+  templateUrl: './change-author-user-dropdown.component.html',
+  styleUrls: ['./change-author-user-dropdown.component.scss'],
+})
+export class ChangeAuthorUserDropdownComponent implements OnInit {
+  // array of usernames representing all selected users
+  selectedAuthor: User;
+
+  searchResults: User[] = [];
+
+  // search query string, modeled to the search
+  query: string;
+
+  differ: IterableDiffer<User>;
+  // fires every time an input event occurs on the search input element
+  userSearchInput$: Subject<string> = new Subject();
+  // fires when this component is destroyed
+  destroyed$: Subject<void> = new Subject();
+  // true if dropdown results should be shown, false if they should be hidden
+  showDropdown: boolean;
+  // true if the component is actively querying the services, false otherwise
+  loading: boolean;
+
+
+  // fired when an author is selected
+  @Output() newAuthor: EventEmitter<User> = new EventEmitter();
+
+  constructor(
+    private userService: UserService,
+    private authService: AuthService,
+    private differs: IterableDiffers
+  ) {
+    this.differ = this.differs.find([]).create(null);
+  }
+
+  ngOnInit() {
+    // subscribe to the search input and fire search after debounce
+    this.userSearchInput$
+      .pipe(debounceTime(650), takeUntil(this.destroyed$))
+      .subscribe((val: string) => {
+        this.findUser(val.trim());
+      });
+
+    // TODO is there a better way to do this?
+    // I want two events to occur when this event fires. 1) after a debounce, perform search
+    // 2) immediately (without debounce) show the popup
```

Comment:
Can I get a little more context on this? We only want to show the drop down if there are users to show correct?

Reply:
This comment was in the original user dropdown component, So Nick will likely be able to give more context on it.

Reply:
Gotcha. Thanks